agatha/construct/semrep_util.py

```diff
@@ -321,15 +423,28 @@ def run(self, input_path:Path, output_path:Path)->None:
     input_path = Path(input_path)
     assert input_path.is_file(), f"Failed to find {input_path}"
     assert not output_path.exists(), f"Refusing to overwrite {output_path}"
-    cmd = self._get_flags(input_path, output_path)
-    env = self._get_env()
-    print("Running:", " ".join(cmd))
-    subprocess.run(
-        cmd,
-        env=env,
-        check=True,
-    )
-    assert output_path.is_file(), f"SemRep Failed to produce {output_path}"
+    if line_mode:
+      buffer = []
+      lines = self._read_lines(input_path)
+      for line in lines:
+        try:
+          buffer += self._run_gracefully([line])
+        except ChildProcessError:
+          print("Caught exception, attempting to restart...")
+          # not sure how to make the jump to the next input
+        finally:
+          # flush output irrespective of exception to output
+          self._write_xml(buffer, output_path)
```

Comment:
I though `_write_xml` was supposed to read lxml elements? Here you're passing it strings?